packages/compass-aggregations/src/components/pipeline-toolbar/pipeline-header/index.tsx

```diff
@@ -0,0 +1,80 @@
+import React from 'react';
+import { css, spacing, Body, Icon } from '@mongodb-js/compass-components';
+import { connect } from 'react-redux';
+import type { ConnectedProps } from 'react-redux';
+import type { Dispatch } from 'redux';
+
+import PipelineStages from './pipeline-stages';
+import PipelineActions from './pipeline-actions';
+import {
+  getSavedPipelines,
+  savedPipelinesListToggle,
+} from '../../../modules/saved-pipeline';
+
+const containerStyles = css({
+  display: 'grid',
+  gap: spacing[4],
+  gridTemplateAreas: '"pipelineTextAndOpen pipelineStages pipelineAction"',
+  gridTemplateColumns: 'min-content',
```

Comment:
This only applies size restrictions to the first column here, you need to either grid auto columns or explicitly specify size for every column here. Without this you have two columns now that try to get auto sized and this looks like this:

<img width="878" alt="image" src="https://user-images.githubusercontent.com/5036933/159236384-60166bbb-e297-416d-b0cb-4fed6cd053b0.png">